src/crypto/mod.rs

```diff
@@ -596,21 +596,21 @@ impl Serialize for COSEKey {
             COSEKeyType::OKP(key) => {
                 map.serialize_entry(&1, &COSEKeyTypeId::OKP)?;
                 map.serialize_entry(&3, &self.alg)?;
-                map.serialize_entry(&-1, &key.curve)?;
                 map.serialize_entry(&-2, &key.x)?;
+                map.serialize_entry(&-1, &key.curve)?;
```

Comment:
I'm not sure this is right. `-2` is `0b001_00001`, but `-1` is `0b001_00000`, which sorts before, right?